shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,201 @@
+            std::vector<double> doubleVector = resolve_expression<double>(testValues["doubles"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["doubles"].size() == doubleVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["doubles"].size(); i++) {
+                    Approx epsilon_result = Approx(doubleVector.at(i)).epsilon(ERROR_THRESHOLD);
+                    REQUIRE(testValues["doubles"][i].as<double>() == epsilon_result);
+                }
+            }
+        }
+
+        // Vector2d
+        AND_WHEN("A vector is declared using Vector2d values from the YAML node") {
+            std::vector<Eigen::Vector2d> eigenV2dVector = resolve_expression<Eigen::Vector2d>(testValues["vector2"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["vector2"].size() == eigenV2dVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["vector2"].size(); i++) {
+                    Approx epsilon_x = Approx(eigenV2dVector.at(i).x()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_y = Approx(eigenV2dVector.at(i).y()).epsilon(ERROR_THRESHOLD);
+                    // NOTE: Must test individual xy values
+                    REQUIRE(testValues["vector2"][i][0].as<double>() == epsilon_x);
+                    REQUIRE(testValues["vector2"][i][1].as<double>() == epsilon_y);
+                }
+            }
+        }
+
+        // Vector3d
+        AND_WHEN("A vector is declared using Vector3d values from the YAML node") {
+            std::vector<Eigen::Vector3d> eigenV3dVector = resolve_expression<Eigen::Vector3d>(testValues["vector3"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(testValues["vector3"].size() == eigenV3dVector.size());
+            }
+            AND_THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < testValues["vector3"].size(); i++) {
+                    Approx epsilon_x = Approx(eigenV3dVector.at(i).x()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_y = Approx(eigenV3dVector.at(i).y()).epsilon(ERROR_THRESHOLD);
+                    Approx epsilon_z = Approx(eigenV3dVector.at(i).z()).epsilon(ERROR_THRESHOLD);
+                    // NOTE: Must test individual xyz values
+                    REQUIRE(testValues["vector3"][i][0].as<double>() == epsilon_x);
+                    REQUIRE(testValues["vector3"][i][1].as<double>() == epsilon_y);
+                    REQUIRE(testValues["vector3"][i][2].as<double>() == epsilon_z);
+                }
+            }
+        }
+
+        // Array conversions
+        // ints
+        AND_WHEN("An array is declared using integers from the YAML node") {
```

Comment:
I'll stop suggesting these :p I think you get the drill
```suggestion
        WHEN("An array is declared using integers from the YAML node") {
```